nurseconnect/views.py

```diff
@@ -31,3 +37,50 @@ def search(request, results_per_page=10):
         "search_results": search_results,
         "results": results,
     })
+
+
+class NurseConnectRegistrationView(RegistrationView):
+    def form_valid(self, form):
+        username = form.cleaned_data['username']
+        password = form.cleaned_data['password']
+        mobile_number = form.cleaned_data['mobile_number']
+        user = User.objects.create_user(username=username, password=password)
+        user.profile.mobile_number = mobile_number
+        user.profile.first_name = ""
```

Comment:
`first_name` and `last_name` are not part of `UserProfile` model, they are part of django's `User` model. You don't have to specify them here. They are created by default.